File: two-pointers/medium/two-sum-ii-input-array-is-sorted/sol.cs
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

public class Solution
{
  public int[] TwoSum(int[] numbers, int target)
  {
    int l = 0, r = numbers.Length - 1;

    while (l < r)
    {
      if (numbers[l] + numbers[r] < target)
      {
        l++;
      }
      else if (numbers[l] + numbers[r] > target)
      {
        r--;
      }
      else
      {
        break;
      }
    }

    return [l + 1, r + 1];
  }
}

File: two-pointers/medium/two-sum-ii-input-array-is-sorted/sol.js
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let l = 0,
    r = numbers.length - 1;

  while (l < r) {
    if (numbers[l] + numbers[r] < target) {
      l++;
    } else if (numbers[l] + numbers[r] > target) {
      r--;
    } else {
      return [l + 1, r + 1];
    }
  }
};
